packages/headless/src/features/commerce/recent-queries/recent-queries-slice.test.ts

```diff
@@ -29,24 +29,57 @@ describe('commerce/recent-queries-slice', () => {
     });
   });
 
-  it('#registerRecentQueries should set queries and maxLength params in state', () => {
+  it('#registerRecentQueries should trim and lowercase queries, and eliminate duplicates in state', () => {
+    state = recentQueriesReducer(
+      state,
+      registerRecentQueries({
+        queries: [
+          'what is LOVE',
+          'Oh baby',
+          "don't hurt me",
+          "   DON'T HURT ME   ",
+          'no more!',
+        ],
+        maxLength: testMaxLength,
+      })
+    );
+
+    expect(state.queries).toEqual([
+      'what is love',
+      'oh baby',
+      "don't hurt me",
+      'no more!',
+    ]);
+  });
+
+  it('#registerRecentQueries should set maxLength params in state', () => {
     state = recentQueriesReducer(
       state,
       registerRecentQueries({queries: testQueries, maxLength: testMaxLength})
     );
 
-    expect(state.queries).toEqual(testQueries);
     expect(state.maxLength).toEqual(testMaxLength);
   });
 
-  it('#registerRecentQueries should shorten queries if it exceeds maxLength param', () => {
-    const queries = ['q1', 'q2', 'q3', 'q4'];
+  it('#registerRecentQueries should only keep queries up to the specified maxLength (after eliminating duplicates) in state', () => {
```

Comment:
Does this actually check that we keep queries up to the max length? After lowercasing and removing duplicates, we have 4 queries, so the max length is never applied to the list, right?